nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -640,14 +635,38 @@ in
       preStart =
         ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
+        ${cfg.package}/bin/nginx -c '${configFile}' -p '/var/lib/${cfg.stateDirName}' -t
         '';
       serviceConfig = {
-        ExecStart = "${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir}";
+        ExecStart = "${cfg.package}/bin/nginx -c '${configFile}' -p '/var/lib/${cfg.stateDirName}'";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
         Restart = "always";
         RestartSec = "10s";
         StartLimitInterval = "1min";
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Filesystem access
+        ProtectSystem = "strict";
```

Comment:
This will break existing cache setups, including mine. I keep my cache in a folder under `/var/cache` (for no particular technical reason, but because it seemed like the proper place). One solution is to set `CacheDirectory="nginx"` and make users either place their caches in `/var/cache/nginx` or add another `CacheDirectory` to the service themselves. That seems better than putting caches in the state directory.

Reply:
I usually use socket files in `/var/run` to communicate with services that I am reverse proxying with nginx and I believe this will break this functionality.